RevenueCatUI/Purchasing/PurchaseHandler.swift

```diff
@@ -58,44 +74,87 @@ final class PurchaseHandler: ObservableObject {
+                     performRestore: PerformRestore? = nil) {
+        self.init(isConfigured: true,
+                  purchases: purchases,
+                  performPurchase: performPurchase,
+                  performRestore: performRestore)
     }
 
     init(
         isConfigured: Bool = true,
-        purchases: PaywallPurchasesType
+        purchases: PaywallPurchasesType,
+        performPurchase: PerformPurchase? = nil,
+        performRestore: PerformRestore? = nil
     ) {
         self.isConfigured = isConfigured
         self.purchases = purchases
+        self.performPurchase = performPurchase
+        self.performRestore = performRestore
+    }
+
+    /// Returns a new instance of `PurchaseHandler` using `Purchases.shared` if `Purchases`
+    /// has been configured, and using a PurchaseHandler that cannot be used for purchases otherwise.
+    public static func `default`(performPurchase: PerformPurchase?,
+                                 performRestore: PerformRestore?) -> Self {
+        return Purchases.isConfigured ? .init(performPurchase: performPurchase,
+                                              performRestore: performRestore) :
+                                        Self.notConfigured(performPurchase: performPurchase,
+                                                           performRestore: performRestore)
     }
 
-    // @PublicForExternalTesting
-    static func `default`() -> Self {
-        return Purchases.isConfigured ? .init() : Self.notConfigured()
+    static func `default`(performPurchase: PerformPurchase?,
+                          performRestore: PerformRestore?,
+                          customerInfo: CustomerInfo?) -> Self {
+        return Purchases.isConfigured ? .init(performPurchase: performPurchase,
+                                              performRestore: performRestore) :
+                                        Self.notConfigured(performPurchase: performPurchase,
+                                                           performRestore: performRestore,
+                                                           customerInfo: customerInfo)
     }
 
-    private static func notConfigured() -> Self {
-        return .init(isConfigured: false, purchases: NotConfiguredPurchases())
+    private static func notConfigured(performPurchase: PerformPurchase?,
+                                      performRestore: PerformRestore?,
+                                      customerInfo: CustomerInfo? = nil) -> Self {
+        return .init(isConfigured: false,
+                     purchases: NotConfiguredPurchases(customerInfo: customerInfo),
+                     performPurchase: performPurchase,
+                     performRestore: performRestore)
     }
 
 }
 
 @available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
 extension PurchaseHandler {
 
+    // MARK: - Purchase
+
     @MainActor
-    func purchase(package: Package) async throws -> PurchaseResultData {
```

Comment:
This was never used anywhere, and we can't create one from the `performExternalPurchaseLogic`, so forget it?

Reply:
The entire `PurchaseHandler` is internal, so nothing here is publicly accessible.

Reply:
I changed things so we do construct one, although Transaction is `nil` when we make an external purchase, unsure if that's unexpected. But as I wrote earlier, nothing uses this....

Reply:
Done!